src/tasks/RuntimeConfigParser/RuntimeConfigParser.cs

```diff
@@ -0,0 +1,113 @@
+    /// </summary>
+    [Required]
+    public string OutputFile { get; set; } = "";
+
+    /// <summary>
+    /// List of properties reserved for the host.
+    /// </summary>
+    public ITaskItem[] ReservedProperties { get; set; } = Array.Empty<ITaskItem>();
+
+    public override bool Execute()
+    {
+        if (string.IsNullOrEmpty(RuntimeConfigFile))
+        {
+            throw new ArgumentException($"'{nameof(RuntimeConfigFile)}' is required.", nameof(RuntimeConfigFile));
+        }
+
+        if (string.IsNullOrEmpty(OutputFile))
+        {
+            throw new ArgumentException($"'{nameof(OutputFile)}' is required.", nameof(OutputFile));
+        }
+
+        Dictionary<string, string> configProperties = ConvertInputToDictionary(RuntimeConfigFile);
+
+        if (ReservedProperties.Length != 0)
+        {
+            checkDuplicateProperties(configProperties, ReservedProperties);
+        }
+
+        var blobBuilder = new BlobBuilder();
+        ConvertDictionaryToBlob(configProperties, blobBuilder);
+
+        using var stream = File.OpenWrite(OutputFile);
+        blobBuilder.WriteContentTo(stream);
+
+        return true;
+    }
+
+    /// Reads a json file from the given path and extracts the "configProperties" key (assumed to be a string to string dictionary)
+    private Dictionary<string, string> ConvertInputToDictionary(string inputFilePath)
+    {
+        var options = new JsonSerializerOptions {
+            AllowTrailingCommas = true,
+            ReadCommentHandling = JsonCommentHandling.Skip,
+        };
+
+        var jsonString = File.ReadAllText(inputFilePath);
+        var parsedJson = JsonSerializer.Deserialize<Root>(jsonString, options);
+
+        if (parsedJson == null)
+        {
+            throw new ArgumentException("Wasn't able to parse the json file successfully.");
+        }
+
+        return parsedJson.ConfigProperties;
+    }
+
+    /// Just write the dictionary out to a blob as a count followed by
+    /// a length-prefixed UTF8 encoding of each key and value
+    private void ConvertDictionaryToBlob(IReadOnlyDictionary<string, string> properties, BlobBuilder builder)
+    {
+        int count = properties.Count;
+
+        builder.WriteCompressedInteger(count);
+        foreach (var kvp in properties)
+        {
+            builder.WriteSerializedString(kvp.Key);
+            builder.WriteSerializedString(kvp.Value);
+        }
+    }
+
+    private void checkDuplicateProperties(IReadOnlyDictionary<string, string> properties, ITaskItem[] keys)
+    {
+        foreach (var key in keys)
+        {
+            if (properties.ContainsKey(key.ItemSpec))
+            {
+                throw new ArgumentException($"Property '{key}' can't be set by the user!");
+            }
+        }
+    }
+}
+
+public class Root
+{
+    // the configProperties key
+    [JsonPropertyName("configProperties")]
+    public Dictionary<string, string> ConfigProperties { get; set; } = new Dictionary<string, string>();
```

Comment:
nit, consider providing comparer explicitly so it's clear you intended case sensitive.

Reply:
To keep the code simple, I decide to leave it as it is.